realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -162,6 +162,9 @@ public void removeChangeListener(RealmChangeListener<? extends BaseRealm> listen
      * }
      * </pre>
      *
+     * <p>Note that when the Realm is accessed from threads other than where it was created, {@link IllegalStateException}
+     * will be thrown. Please avoid using {@code subscribeOn()}, and use {@code Realm.where().*Async()} methods instead.
```

Comment:
Actually it is the combination of observeOn/subscribeOn that is problematic, since that will cause the thread changes, so I would probably rephrase this to:

"Care should  be taken when using different schedulers with {@code subscribeOn()} and {@code observeOn}. Consider using {@code Realm.where().find*Async()} instead."